apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/preadj/FissClaimTransformerV2.java

```diff
@@ -112,132 +196,279 @@ private static CodeableConcept getPriority() {
   }
 
   private static Money getTotal(PreAdjFissClaim claimGroup) {
-    Money total = new Money();
+    Money total;
 
-    total.setValue(claimGroup.getTotalChargeAmount());
-    total.setCurrency("USD");
+    if (claimGroup.getTotalChargeAmount() != null) {
+      total = new Money();
+
+      total.setValue(claimGroup.getTotalChargeAmount());
+      total.setCurrency("USD");
+    } else {
+      total = null;
+    }
 
     return total;
   }
 
-  private static List<Resource> getContainedProvider(PreAdjFissClaim claimGroup) {
-    return Collections.singletonList(
-        new Organization()
+  private static Resource getContainedPatient(PreAdjFissClaim claimGroup) {
+    Optional<PreAdjFissPayer> optional =
+        claimGroup.getPayers().stream()
+            .filter(p -> p.getPayerType() == PreAdjFissPayer.PayerType.BeneZ)
+            .findFirst();
```

Comment:
Just checking, we only expect one patient per this logic? Since we're using findFirst, we'd ignore any beyond the first.

Reply:
Yes, I've been made to understand that we will only ever have one payer of type `BeneZ`.